tracer.go

```diff
@@ -45,20 +45,41 @@ const (
 )
 
 var (
-	// DefaultTracer is the default global Tracer, set at package
-	// initialization time, configured via environment variables.
-	//
-	// This will always be initialized to a non-nil value. If any
-	// of the environment variables are invalid, the corresponding
-	// errors will be logged to stderr and the default values will
-	// be used instead.
-	DefaultTracer *Tracer
+	tracerMu      sync.RWMutex
+	defaultTracer *Tracer
 )
 
-func init() {
-	var opts TracerOptions
-	opts.initDefaults(true)
-	DefaultTracer = newTracer(opts)
+// DefaultTracer returns the default global Tracer, set the first time the
+// function is called. It is configured via environment variables.
+//
+// This will always be initialized to a non-nil value. If any of the
+// environment variables are invalid, the corresponding errors will be logged
+// to stderr and the default values will be used instead.
+func DefaultTracer() *Tracer {
+	tracerMu.RLock()
+	dt := defaultTracer
+	tracerMu.RUnlock()
+
+	if dt == nil {
+		var opts TracerOptions
+		opts.initDefaults(true)
+		SetDefaultTracer(newTracer(opts))
+	}
+	return defaultTracer
```

Comment:
this is accessing `defaultTracer` without a lock, which is racy

I think it should be:

```go
	tracerMu.RLock()
	tracer := defaultTracer
	tracerMu.RUnlock()
	if tracer != nil {
		return tracer
	}

	var opts TracerOptions
	opts.initDefaults(true)
	tracer = newTracer(opts)
	SetDefaultTracer(tracer)
	return tracer
```